src/mechanics/constitutive/ModifiedMises.h

```diff
@@ -0,0 +1,253 @@
+#pragma once
+#include "base/Exception.h"
+#include "mechanics/constitutive/ConstitutivePlaneStateEnum.h"
+#include "mechanics/constitutive/EngineeringStrain.h"
+#include "mechanics/constitutive/EngineeringStrainInvariants.h"
+
+namespace NuTo
+{
+namespace Constitutive
+{
+/**
+ * Equivalent strain (strain norm) based on the modified mises norm.
+ * De Vree et al. "Comparision of nonlocal approaches in continuum damage mechanics", Computers \& Structures,
+ * 1995.
+ * https://doi.org/10.1016/0045-7949(94)00501-S
+ *
+ * It includes the value `k` in a way that expresses the ratio of the materials compressive to tensile strength:
+ * A
+ * uniaxial tensile strain and a k-times higher uniaxial compressive strain both lead to the same equivalent
+ * strain.
+ * With the strain tensor invariant I1 and the deviatoric strain tensor invariant J2, it reads
+ * \f[/
+ * \varepsilon_\text{eq}(\boldsymbol \varepsilon) = \frac{k-1}{2k(1-2\nu)}I_1 +
+ * \frac{1}{2k}\sqrt{\left(\frac{k-1}{1-2\nu}I_1\right)^2 + \frac{2k}{(1+\nu)^2}J_2}
+ * \f]
+ *
+ * @tparam TDim dimension 1,2,3
+ */
+template <int TDim>
+class ModifiedMises
+{
+public:
+    //! Constructor.
+    //! @param nu Poissons ratio
+    //! @param k ratio of compressive strength to tensile strength, ~10 for concrete
+    //! @param planeState PLANE_STRAIN or PLANE_STRESS
+    ModifiedMises(double nu, double k, ePlaneState planeState = ePlaneState::PLANE_STRAIN);
+
+    //! @param strain strain to evaluate
+    //! @return the value of the modified Mises strain norm
+    double Value(const NuTo::EngineeringStrain<TDim>& strain) const;
+
+    //! @param strain strain to evaluate
+    //! @return the derivative of the modified Mises strain norm with respect to the strain
+    Eigen::Matrix<double, Voigt::Dim(TDim), 1> Derivative(const NuTo::EngineeringStrain<TDim>& strain) const;
+
+private:
+    //! This method may be useful for other classes. However, it cannot be a member of
```

Comment:
Imho, this shouldn't be in Doxygen. I'd like to know what `Strain3D` does, not whether it may be useful for other classes. Let's make that either regular comments or `@remark`s.